frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.unit.spec.tsx

```diff
@@ -55,18 +49,26 @@ const TestWrapper = ({
   );
 };
 
-const setup = ({ initialParameter, otherParameters }: SetupOpts) => {
-  const setParameterMock = jest.fn();
-
+const setup = ({
+  initialParameter,
+  nextParameter,
+  otherParameters,
+}: SetupOpts): {
+  clickNextParameterButton: () => void;
+} => {
   renderWithProviders(
     <TestWrapper
       initialParameter={initialParameter}
+      nextParameter={nextParameter}
       otherParameters={otherParameters}
-      onSetParameter={setParameterMock}
     />,
   );
 
-  return { setParameterMock };
+  // clicking the next parameter button will update the parameter from initialParameter to nextParameter
+  return {
+    clickNextParameterButton: () =>
+      userEvent.click(screen.getByTestId("parameter-sidebar-test-change")),
```

Comment:
@oisincoveney I introduced this indirection to not require tests to know about the test ID. Only the setup function needs to know about how to click the button

Reply:
I think you might be able to split this into its own function - the setup function doesn't even need to know how to click the button

Reply:
Yes, but I think putting TestWrapper back inside `setup` resolves the conflict - now setup's implementation explicitly knows how to click the button, and nothing else does. https://github.com/metabase/metabase/pull/34613/files/4fbb73408784e52bef448b4ca1c2329a3a85f8cc..c9bd641a7751e22f181f1e1b6be905854eb4c272